src/scripts/import-open-api.ts

```diff
@@ -782,43 +924,54 @@ const importOpenApi = async ({
   Object.entries(specs.paths).forEach(([route, verbs]: [string, PathItemObject]) => {
     Object.entries(verbs).forEach(([verb, operation]: [string, OperationObject]) => {
       if (["get", "post", "patch", "put", "delete"].includes(verb)) {
-        output += generateRestfulComponent(
-          operation,
-          verb,
-          route,
-          operationIds,
-          verbs.parameters,
-          specs.components,
-          customProps,
-          customGenerator,
-        );
+        if (ts) {
+          output += generateTypescriptApi(
+            verb,
+            operation.responses,
+            route,
+            operation.operationId,
+            operation.requestBody,
+            operation.parameters,
+          );
+        } else {
+          output += generateRestfulComponent(
```

Comment:
This looks like of a lot of duplication over there, I will be more in favor of adding a param to `generateRestfulComponent` to skip the jsx (and this is maybe a goodtime to merge the param under an `options` object to avoid mistake)